src/parser/test/test.ml

```diff
@@ -0,0 +1,9262 @@
+         (warnings
+          ( "File \"f.ml\", line 1, characters 6-7:\
+           \n'-' (bulleted list item) should begin on its own line."
+            "File \"f.ml\", line 1, characters 6-7:\
+           \n'-' (bulleted list item) is not allowed in the tags section.\
+           \nSuggestion: move '-' (bulleted list item) before any tags."))) |}]
+  end
+  [@ocaml.warning "-32"] in
+  ()
+
+let%expect_test _ =
+  let module Closed = struct
+    let basic =
+      test "@closed";
+      [%expect {| ((output (((f.ml (1 0) (1 7)) @closed))) (warnings ())) |}]
+
+    let prefix =
+      test "@closedfoo";
+      [%expect
+        {|
+        ((output
+          (((f.ml (1 0) (1 10))
+            (paragraph (((f.ml (1 0) (1 10)) (word @closedfoo)))))))
+         (warnings
+          ( "File \"f.ml\", line 1, characters 0-10:\
+           \nUnknown tag '@closedfoo'."))) |}]
+
+    let extra_whitespace =
+      test "@closed";
+      [%expect {| ((output (((f.ml (1 0) (1 7)) @closed))) (warnings ())) |}]
+
+    let followed_by_junk =
+      test "@closed foo";
+      [%expect
+        {|
+        ((output
+          (((f.ml (1 0) (1 7)) @closed)
+           ((f.ml (1 8) (1 11)) (paragraph (((f.ml (1 8) (1 11)) (word foo)))))))
+         (warnings
+          ( "File \"f.ml\", line 1, characters 8-11:\
+           \nParagraph is not allowed in the tags section.\
+           \nSuggestion: move 'foo' before any tags."
+            "File \"f.ml\", line 1, characters 8-11:\
+           \nParagraph should begin on its own line."))) |}]
+
+    let followed_by_paragraph =
+      test "@closed\nfoo";
+      [%expect
+        {|
+        ((output
+          (((f.ml (1 0) (1 7)) @closed)
+           ((f.ml (2 0) (2 3)) (paragraph (((f.ml (2 0) (2 3)) (word foo)))))))
+         (warnings
+          ( "File \"f.ml\", line 2, characters 0-3:\
+           \nParagraph is not allowed in the tags section.\
+           \nSuggestion: move 'foo' before any tags."))) |}]
+
+    let followed_by_tag =
+      test "@closed\n@deprecated";
+      [%expect
+        {|
+        ((output (((f.ml (1 0) (1 7)) @closed) ((f.ml (2 0) (2 11)) (@deprecated))))
+         (warnings ())) |}]
+
+    let with_list =
+      test "@closed - foo";
+      [%expect
+        {|
+        ((output
+          (((f.ml (1 0) (1 7)) @closed)
+           ((f.ml (1 8) (1 13))
+            (unordered light
+             ((((f.ml (1 10) (1 13)) (paragraph (((f.ml (1 10) (1 13)) (word foo)))))))))))
+         (warnings
+          ( "File \"f.ml\", line 1, characters 8-9:\
+           \n'-' (bulleted list item) should begin on its own line."
+            "File \"f.ml\", line 1, characters 8-9:\
+           \n'-' (bulleted list item) is not allowed in the tags section.\
+           \nSuggestion: move '-' (bulleted list item) before any tags."))) |}]
+  end
+  [@ocaml.warning "-32"] in
+  ()
+
+let%expect_test _ =
+  let module Reference_component_kind = struct
```

Comment:
This section is meaningless, for the parser, references are just strings. This is meant to test `Odoc_model.Reference.parse`.
I suggest:
- Copy the tests related to quotes and a few tests related to the `kind-` syntax to the `Simple_reference` section
  (the `kind-` syntax is not tested there, only `kind:` is)
- Move all these tests to the main Odoc project (in a new `test/model` directory)